dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,141 @@
+package com.dotcms.filters.interceptor.saml;
+
+import com.dotcms.cms.login.LoginServiceAPI;
+import com.dotcms.filters.interceptor.Result;
+import com.dotcms.filters.interceptor.WebInterceptor;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.web.UserWebAPI;
+import com.dotmarketing.cms.factories.PublicEncryptionFactory;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.PortalException;
+import com.liferay.portal.SystemException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+/**
+ * This interceptor encapsulates the logic for Saml
+ * Basically if there is any configuration set on apps and c
+ * @author jsanca
+ */
+public class SamlWebInterceptor implements WebInterceptor {
+
+    protected static final String BY_PASS_KEY   = "native";
+    protected static final String BY_PASS_VALUE = "true";
+
+    private final LoginServiceAPI loginServiceAPI;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'loginServiceAPI' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)